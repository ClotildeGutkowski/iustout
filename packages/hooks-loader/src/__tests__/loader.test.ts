import { resolve } from 'path'
import compiler from './compiler'
import webpack from 'webpack'

const resolveEntry = (path: string) => {
  return resolve(__dirname, './fixtures/src/apis/lambda', path)
}

const getOutput = (stats: webpack.Stats) => {
  return stats.toJson().modules[2].source
}

test('Compile normally exported functions', async () => {
  const stats = await compiler(resolveEntry('index.ts'))
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile nest function', async () => {
  const stats = await compiler(resolveEntry('nest/index.ts'))
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile export default arrow', async () => {
  const stats = await compiler(resolveEntry('arrow.ts'))
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile export types', async () => {
  const stats = await compiler(resolveEntry('type.ts'))
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile type assert', async () => {
  const stats = await compiler(resolveEntry('type-assert.ts'))
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile params', async () => {
  const stats = await compiler(resolveEntry('param.ts'))
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile error export', async () => {
  const fixture = resolveEntry('error.ts')
  try {
    await compiler(fixture)
  } catch (error) {
    expect(error).toBeTruthy()
  }
})
