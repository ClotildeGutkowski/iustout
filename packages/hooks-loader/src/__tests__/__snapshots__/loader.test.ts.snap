// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compile export default arrow 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  default  function $default (...args) {
  return request({
    url: '/api/arrow',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"arrow\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;

exports[`Compile export types 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  function normal (...args) {
  return request({
    url: '/api/type/_normal',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"type-normal\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;

exports[`Compile nest function 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  function nestFunc (...args) {
  return request({
    url: '/api/nest/_nestFunc',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"nest-index-nestfunc\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;

exports[`Compile normally exported functions 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  function variableFuncExpression (...args) {
  return request({
    url: '/api/_variableFuncExpression',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablefuncexpression\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function variableFuncExpressionWithArgs (...args) {
  return request({
    url: '/api/_variableFuncExpressionWithArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablefuncexpressionwithargs\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function variableArrowFunction (...args) {
  return request({
    url: '/api/_variableArrowFunction',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablearrowfunction\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function variableArrowFunctionWithArgs (...args) {
  return request({
    url: '/api/_variableArrowFunctionWithArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablearrowfunctionwithargs\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function functionDeclartion (...args) {
  return request({
    url: '/api/_functionDeclartion',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-functiondeclartion\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function functionDeclartionWithArgs (...args) {
  return request({
    url: '/api/_functionDeclartionWithArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-functiondeclartionwithargs\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  default  function hasName (...args) {
  return request({
    url: '/api',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index\\",
  \\"unstable_params\\": [
    \\"args\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function dotdotdotArgs (...args) {
  return request({
    url: '/api/_dotdotdotArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-dotdotdotargs\\",
  \\"unstable_params\\": [
    \\"args\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;

exports[`Compile params 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  function identifier (...args) {
  return request({
    url: '/api/param/_identifier',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-identifier\\",
  \\"unstable_params\\": [
    \\"a\\",
    \\"b\\",
    \\"c\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function assgin (...args) {
  return request({
    url: '/api/param/_assgin',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-assgin\\",
  \\"unstable_params\\": [
    \\"a\\",
    \\"b\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function rest (...args) {
  return request({
    url: '/api/param/_rest',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-rest\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function mix (...args) {
  return request({
    url: '/api/param/_mix',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-mix\\",
  \\"unstable_params\\": [
    \\"a\\",
    \\"b\\",
    \\"c\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;

exports[`Compile type assert 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  function typeAssert (...args) {
  return request({
    url: '/api/type-assert/_typeAssert',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"type-assert-typeassert\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;
